dev/io.openliberty.org.jboss.resteasy.mprestclient.4.0/src/org/jboss/resteasy/microprofile/client/ot/LibertyRestClientBuilderImpl.java

```diff
@@ -942,12 +951,19 @@ private static Method[] resolveMethods(final Class<?> type) {
     }
 
     // Liberty Change Start
-    private static ClassLoader getClassLoader(Class<?> clazz) {
+    private ClassLoader getClassLoader(Class<?> clazz) {
+        ClassLoader clazzLoader = null;
         if (System.getSecurityManager() == null) {
-            return clazz.getClassLoader();
+            clazzLoader = clazz.getClassLoader();
+        } else {
+            clazzLoader = AccessController.doPrivileged((PrivilegedAction<ClassLoader>) clazz::getClassLoader);
         }
-        return AccessController.doPrivileged((PrivilegedAction<ClassLoader>) clazz::getClassLoader);
+
+        myClassLoader.addLoader(clazzLoader);
```

Comment:
addLoader() doesn't check to see if the parent matches the passed in ClassLoader.  As such I think the logic should be updated to:

```suggestion
        if (clazzLoader != LibertyRestClientBuilderImpl.class.getClassLoader()) {
            myClassLoader.addLoader(clazzLoader);
        }
```

Of course getClassLoader() requires a doPriv.  Can update to get the LibertyRestClientBuilderImpl's classloader in static initializer with doPriv around it and store it in a static variable in the class and then you don't need to get it with a doPriv in the constructor or here.